src/ConditionsResolver/Resovler/ConditionalGroup.cs

```diff
@@ -0,0 +1,326 @@
+
+        private string[] ParseOrs(string input) => FindOrs.SplitOnce(input);
+
+        private string[] ParseAnds(string input) => FindAnds.SplitOnce(input);
+
+        public void Set(string left, string right, Keyword keyword)
+        {
+            Keyword = keyword;
+            if (!string.IsNullOrEmpty(left))
+            {
+                if (FindAnds.Matches(left).Any() || FindOrs.Matches(left).Any())
+                {
+                    LeftGroup = ConditionalGroup.Create(left, GroupedLogical);
+                }
+                else
+                {
+                    LeftConditional = Conditional.Create(left);
+                }
+            }
+            if (!string.IsNullOrEmpty(right))
+            {
+                if (FindAnds.Matches(right).Any() || FindOrs.Matches(right).Any() || FindBrackets.Matches(right).Any())
+                {
+                    RightGroup = ConditionalGroup.Create(right, GroupedLogical);
+                }
+                else
+                {
+                    RightConditional = Conditional.Create(right);
+                }
+            }
+        }
+
+        public void Parse(string input, int groupedLogicalParent = 0)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                throw new ArgumentNullException(nameof(input));
+            }
+
+            var foundOpenBrackets = FindBrackets.Matches(input);
+            var foundClosingBrackets = FindCloseBrackets.Matches(input);
+            if (GroupedLogical > 1 && foundOpenBrackets.Count != foundClosingBrackets.Count)
+            {
+                throw new ArgumentException("Matching brackets missing.");
+            }
+
+            var foundAnds = FindAnds.Matches(input);
+            var foundOrs = FindOrs.Matches(input);
+
+            if (foundOpenBrackets.Count >= 1)
+            {
+                var firstOpenBracketIndex = foundOpenBrackets.First().Index;
+                if (firstOpenBracketIndex == 0)
+                {
+                    GroupedLogical = groupedLogicalParent + 1;
+                    input = input.Trim('(');
+                }
+            }
+
+            if (foundAnds.Count == 1 && foundOrs.Count == 0)
+            {
+                var splitByAnd = ParseAnds(input);
+                Set(splitByAnd[0], splitByAnd[1].TrimStartExt("AND"), Keyword.AND);
+                return;
+            }
+            if (foundAnds.Count == 0 && foundOrs.Count == 1)
+            {
+                var splitByOr = ParseOrs(input);
+                Set(splitByOr[0], splitByOr[1].TrimStartExt("OR"), Keyword.OR);
+                return;
+            }
+            if (foundAnds.Count == 0 && foundOrs.Count == 0)
+            {
+                Set(input, string.Empty, Keyword.SINGULAR);
+                return;
+            }
+
+            ParseComplex(input);
+        }
+
+        public void ParseBrackets(string input)
+        {
+            if (string.IsNullOrEmpty(input))
```

Comment:
FYI - there is a function that checks if the string contains only spaces.
We also reference Ardalis.GuardClauses to simplify null, null or empty, null or whitespace checks.